src/request.ts

```diff
@@ -34,25 +36,107 @@ async function startFetchRequest(
   emitter: EventEmitter,
   counter: number = 0,
 ) {
+  const logger = getLogger('fetch');
   const { httpProxy, followRedirect } = options;
   const agent = httpProxy ? createHttpsProxyAgent(httpProxy) : undefined;
   const { url, body, ...rrequest } = request;
   const controller = new AbortController();
-  let res;
+
+  let retryCount = 0;
+
+  const retryOpts: Required<HttpRequestOptions['retry']> = {
+    maxRetries: options.retry?.maxRetries ?? 5,
+    errorCodes: options.retry?.errorCodes ?? [
+      'ECONNRESET',
+      'ECONNREFUSED',
+      'ENOTFOUND',
+      'ENETDOWN',
+      'ENETUNREACH',
+      'EHOSTDOWN',
+      'UND_ERR_SOCKET',
+      'ETIMEDOUT',
+      'EPIPE',
+    ],
+    methods: options.retry?.methods ?? [
+      'GET',
+      'PUT',
+      'HEAD',
+      'OPTIONS',
+      'DELETE',
+    ],
+  };
```

Comment:
these are the default errorCodes and methods in node's fetch retry handler:
https://undici.nodejs.org/#/docs/api/RetryAgent?id=parameter-retryhandleroptions

plus these 2: `ETIMEDOUT` and `EPIPE`